test/unit/swap-proxy/SwapProxy.t.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.22;
+
+import { ISwapProxy, SwapProxy } from "../../../src/SwapProxy.sol";
+import { PRBTest } from "@prb/test/PRBTest.sol";
+import { StdUtils } from "forge-std/StdUtils.sol";
+import { console2 } from "forge-std/console2.sol";
+import { ERC20MintableMock } from "../../mocks/ERC20/ERC20MintableMock.sol";
+import { UniversalPermit2Adapter } from "@mean-finance/permit2-adapter/src/UniversalPermit2Adapter.sol";
+import { MockPermit2 } from "@mean-finance/permit2-adapter/test/unit/mocks/MockPermit2.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract SwapProxyTest is PRBTest, StdUtils {
+  ISwapProxy private swapProxy;
+  UniversalPermit2Adapter underlyingSwapper;
+  ERC20MintableMock private token = new ERC20MintableMock();
+
+  function setUp() public virtual {
+    underlyingSwapper = new UniversalPermit2Adapter(new MockPermit2());
```

Comment:
Instead of using the Permit2 adapter here, let's actually build a mock that would allow us to test that:
1. ERC20 tokens are sent
2. Native tokens are sent
3. Function is called correctly (via data)
4. Value returned by the underlying swapper is returned by the swap proxy
5. If the underlying swapper reverts, the swap proxy call also reverts